libfreerdp/core/nla.c

```diff
@@ -1129,7 +1129,7 @@ SECURITY_STATUS nla_encrypt_public_key_hash(rdpNla* nla)
 		goto out;
 
 	/* include trailing \0 from hashMagic */
-	if (!winpr_Digest_Update(sha256, hashMagic, strlen(hashMagic) + 1))
+	if (!winpr_Digest_Update(sha256, (const BYTE*)hashMagic, strlen(hashMagic) + 1))
```

Comment:
Why not just declare the buffer as `BYTE` and use `ARRAYSIZE` for the length?

Reply:
and if we are at cleanups, have a look, there are still a lot of broken/wrong/invalid `(int)` casts around